.dotnet/src/Generated/ClientShared/ModelReaderWriterExtensions.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
All the files under Generated/ClientShared will be emitted into client code by the generator.  These are the latest version I believe.  They replace what Azure.Core does with shared source in unbranded clients.  We may or may not want to review them at some point, but they should be very similar to what's in Azure.Core today.